### PR TITLE
[chainerx] Implement Relu in c++

### PR DESCRIPTION
**Pull request by dido1998**
ref:   https://github.com/chainer/chainer/blob/077fdb1616ce957d3b311e28236a43ca2f0a8aa3/chainerx/activation.py#L17

### CODE REVIEW

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 1]
```diff
@@ -1671,6 +1671,32 @@ def forward_xp(self, inputs, xp):
         return xp.sigmoid(inputs[0]),
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'in_dtypes,out_dtype': _in_out_float_dtypes_math_functions,
+        'input': [-2, 2],
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: How about adding `0` for the boundary-value test?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 2]
```diff
@@ -1671,6 +1671,32 @@ def forward_xp(self, inputs, xp):
...
+        'contiguous': [None, 'C'],
+    })
+    # Special values
+    + chainer.testing.product({
+        'shape': [(2, 3)],
+        'in_dtypes,out_dtype': _in_out_float_dtypes_math_functions,
+        'input': [float('inf'), -float('inf'), float('nan')],
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: How about adding `0` for the boundary-value test?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 1 above)
Comment (asi1024) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Sorry, this boundary case is non-differentiable, so it should be added here:
https://github.com/chainer/chainer/pull/6731/files#r271583919

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 2 above)
Comment (dido1998) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Yes, I will add it here.

File: chainerx_cc/chainerx/routines/math.cc [hunk 3]
```diff
@@ -669,6 +669,12 @@ Array Sigmoid(const Array& x) {
     return Reciprocal(1 + Exp(-x_cast));
 }
 
+Array Relu(const Array& x) {
+    Dtype dtype = GetMathResultDtype(x.dtype());
+    const Array& x_cast = x.dtype() == dtype ? x : x.AsType(dtype);
+    return Maximum(0, x_cast);
```
Comment (asi1024) [inline chainerx_cc/chainerx/routines/math.cc]:
```suggestion
    return Maximum(0, x_cast);
```

File: chainerx/activation.py [hunk 4]
```diff
@@ -1,18 +0,0 @@
-import chainerx
-
-
-def relu(x):
-    """Rectified Linear Unit function.
```
Comment (asi1024) [inline chainerx/activation.py]: Could you move this comment to `chainerx/_docs/routines.py`?

**Comment by asi1024** [comment]
@dido1998 Travis test seems to fail for a dtype mismatch. Could you fix it?

**Comment by dido1998** [comment]
I will check it out today.

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 5]
```diff
@@ -1671,6 +1671,32 @@ def forward_xp(self, inputs, xp):
         return xp.sigmoid(inputs[0]),
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'in_dtypes,out_dtype': _in_out_float_dtypes_math_functions,
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Why not testing also with integer dtypes?
```suggestion
        'in_dtypes,out_dtype': _in_out_dtypes_math_functions,
```

**Comment by dido1998** [comment]
I have made the requested changes.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM. Thanks.
